### PR TITLE
setup image hosting using amazon 

### PR DESCRIPTION
**Pull request by Suuuuha**
related to #50

### CODE REVIEW

**Comment by Suuuuha** [comment]
some notes: 
- all the class names will be changed when we have the form tab 
- same for some variables i.e "file", "fileName"

File: app/public/map.js [hunk 1]
```diff
@@ -97,4 +97,46 @@ var nazarethNuggets = (function () { // eslint-disable-line
...
+    const file = files[0]
+    if (file === null) {
+      console.log('No file selected')
+    }
+    getSignedRequest(file)
+  }
+
+  function getSignedRequest (file) {
+    const xhr = new XMLHttpRequest()
+    xhr.open('GET', `/sign-s3?file-name=${file.name}&file-type=${file.type}`)
```
Comment (mattlub) [inline app/public/map.js]: es6...

File: app/public/map.js [hunk 2]
```diff
@@ -97,4 +97,46 @@ var nazarethNuggets = (function () { // eslint-disable-line
   centerButton.addEventListener('click', function (e) {
     mymap.locate({setView: true})
   })
+
+  // Amazon S3
+  document.querySelector('.class-unknown').onchange = function () {
```
Comment (mattlub) [inline app/public/map.js]: make this class name a bit clearer, 'image-input'?

File: app/public/map.js [hunk 3]
```diff
@@ -97,4 +97,46 @@ var nazarethNuggets = (function () { // eslint-disable-line
...
+    const files = document.querySelector('.class-unknown').files
+    const file = files[0]
+    if (file === null) {
+      console.log('No file selected')
```
Comment (mattlub) [inline app/public/map.js]: return here? probably don't want to log this either.

File: app/public/map.js [hunk 4]
```diff
@@ -97,4 +97,46 @@ var nazarethNuggets = (function () { // eslint-disable-line
   centerButton.addEventListener('click', function (e) {
     mymap.locate({setView: true})
   })
+
+  // Amazon S3
+  document.querySelector('.class-unknown').onchange = function () {
+    const files = document.querySelector('.class-unknown').files
+    const file = files[0]
```
Comment (mattlub) [inline app/public/map.js]: what else is in the files array?

File: app/public/map.js [hunk 5]
```diff
@@ -97,4 +97,46 @@ var nazarethNuggets = (function () { // eslint-disable-line
...
+  function getSignedRequest (file) {
+    const xhr = new XMLHttpRequest()
+    xhr.open('GET', `/sign-s3?file-name=${file.name}&file-type=${file.type}`)
+    xhr.onreadystatechange = function () {
+      if (xhr.readyState === 4) {
+        if (xhr.status === 200) {
+          const response = JSON.parse(xhr.responseText)
+          uploadFile(file, response.signedRequest, response.url)
+        } else {
+          console.log('Could not get signed URL.')
```
Comment (mattlub) [inline app/public/map.js]: do something better or add a comment to say something better needs to be done.

File: app/public/map.js [hunk 6]
```diff
@@ -97,4 +97,46 @@ var nazarethNuggets = (function () { // eslint-disable-line
...
+  function uploadFile (file, signedRequest, url) {
+    const xhr = new XMLHttpRequest()
+    xhr.open('PUT', signedRequest)
+    xhr.onreadystatechange = function () {
+      if (xhr.readyState === 4) {
+        if (xhr.status === 200) {
+          // adds the url to a hidden input in the form to send it to the server
+          document.querySelector('.hidden-class-inform').value = url
+        } else {
+          console.log('Could not upload file.')
```
Comment (mattlub) [inline app/public/map.js]: do something better or add a comment to say something better needs to be done.

**Review by mattlub — CHANGES_REQUESTED** [review]

File: app/public/map.js (hunk 4 above)
Comment (Suuuuha) [reply app/public/map.js]: for now theres nothing else cause we uploaded just 1 file

File: app/public/map.js [hunk 7]
```diff
@@ -99,25 +99,25 @@ var nazarethNuggets = (function () { // eslint-disable-line
...
 
   // Amazon S3
-  document.querySelector('.class-unknown').onchange = function () {
-    const files = document.querySelector('.class-unknown').files
+  document.querySelector('.image-input').onchange = function () {
+    const files = document.querySelector('.image-input').files
     const file = files[0]
     if (file === null) {
-      console.log('No file selected')
+      // should send the user that he didn't uplaod a file or it wasn't succeful
```
Comment (mattlub) [inline app/public/map.js]: need to return here or have an else block

**Review by mattlub — CHANGES_REQUESTED** [review]

**Review by mattlub — APPROVED** [review]

**Review by Karyum — APPROVED** [review]

**Pull request merged**